Domain/Game/Game.cs

```diff
@@ -1,24 +1,71 @@
-﻿using Domain.Players;
+﻿using Domain.Exceptions;
+using Domain.Players;
 using Domain.PlayingCards;
 using System;
 using System.Collections.Generic;
 using System.Linq;
 
 namespace Domain.Game
 {
+    [Serializable]
     public class Game
     {
         public string Id { get; }
         public List<Player> Players { get; }
         public Deck<PlayingCard> Deck { get; set; }
         public Stack<PlayingCard> DiscardedCards { get; set; }
 
-        public Game(List<Player> players)
+        private object lockObj;
+        
+        public Game(Player player)
         {
             Id = Guid.NewGuid().ToString();
-            Players = players;
             DiscardedCards = new Stack<PlayingCard>();
             Deck = new Deck<PlayingCard>();
+            Players = new List<Player>();
+            lockObj = new object();
+            Players.Add(player);
+        }
+        
+        public bool JoinPlayer(Player player)
+        {
+            lock(lockObj)
+            {
+                if (Players.Count < 7)
+                {
+                    Players.Add(player);
+                    return true;
+                }
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// removes player from list of players in the game
+        /// </summary>
+        /// <param name="player">player to be removed</param>
+        /// <returns>is players count greater than zero</returns>
+        public bool KickPlayer(Player player)
```

Comment:
Methods JoinPlayer and KickPlayer look the opposite. 
JoinPlayer returns false, if player isn't added And it's OK.
KickPlayer returns false, if there is no player in the game. But it's confusing, because I expect that false means player wasn't removed.

Reply:
done